one_fm/api/doc_methods/payroll_entry.py

```diff
@@ -79,25 +79,45 @@ def set_bank_details(employee_details):
 	Returns:
 		employee_details ([dict): Sets the bank account IBAN code and Bank Code.
 	"""
+
+	employee_missing_detail = []
 	for employee in employee_details:
-		iban, bank, bank_account_no = frappe.db.get_value("Bank Account",{"party":employee.employee},["iban","bank", "bank_account_no"])
-		salary_mode = frappe.db.get_value("Employee", {'name': employee.employee}, ["salary_mode"])
-		if not salary_mode:
-			frappe.throw(_("No salary mode set for {employee}".format(employee=employee.employee)))
-		if not bank:
-			frappe.throw(_("Bank Details for {0} does not exists").format(employee.employee))
-		
-		employee.salary_mode = salary_mode
-		employee.iban_number = iban or bank_account_no
-		bank_code = frappe.db.get_value("Bank", {'name': bank}, ["bank_code"])
-		employee.bank_code = bank_code
+		try:
+			# print(employee, frappe.db.get_value("Bank Account",{"party":employee.employee},["iban","bank", "bank_account_no"]), '\n\n\n')
+			bank_account = frappe.db.get_value("Bank Account",{"party":employee.employee},["iban","bank", "bank_account_no"])
+			salary_mode = frappe.db.get_value("Employee", {'name': employee.employee}, ["salary_mode"])
+			if bank_account:
+				iban, bank, bank_account_no = bank_account
+			else:
+				iban, bank, bank_account_no = None, None, None
+
+			if(salary_mode.lower()=='bank' and bank is None):
+				employee_missing_detail.append({'employee':employee, 'salary_mode':salary_mode, 'issue':'No bank account'})
+			if not salary_mode:
+				employee_missing_detail.append({'employee':employee, 'salary_mode':'', 'issue':'No salary mode'})
+			employee.salary_mode = salary_mode
+			employee.iban_number = iban or bank_account_no
+			bank_code = frappe.db.get_value("Bank", {'name': bank}, ["bank_code"])
+			employee.bank_code = bank_code
+		except Exception as e:
+			frappe.log_error(str(e), 'Payroll Entry')
+			frappe.throw(str(e))
+	# print(employee_missing_detail)

```

Comment:
You can remove this commented print statement

Reply:
I meant, it will fetch the employees normally,  but inform HR about missing salary mode. Before it generates errors in background and fails but now it will process and report as well, what do you think?